### DISCUSSION TITLE
How do I add resources to a popup

### DISCUSSION THREAD

**Discussion by nodoid** [CommunityToolkit/Maui#525] [Q&A]
Given the Popup is an inheritted class (I'd imagine it's based on the content page), there doesn't seem to be a way to set the resources on a Popup page itself (there is no toolkit:Popup.Resources found in intellisense). Am I missing something or does this just not currently exist?

**Comment by bijington** [comment]
It looks like this isn't currently possibly with the `Popup` as it inherits from `Element`. @pictos do you think we could potentially inherit from [`VisualElement`](https://github.com/dotnet/maui/blob/main/src/Controls/src/Core/VisualElement.cs) to allow for the `Resources` to be defined? Or we could implement the [`IResourcesProvider`](https://github.com/dotnet/maui/blob/bf13e62fd699a3d6387f825259668866eddd2419/src/Controls/src/Core/IResourcesProvider.cs#L3) interface although I don't know if it is just as simple as implementing this interface.

**Reply by pictos** [reply]
So... I had that conversation with @PureWeen a while ago, and we can apply styles to `Element` types (like BaseShellItem is an `Element`), If you try to set a style in App.xaml for example, that should work. Anyway, a workaround for that is to apply the style to the content of the Popup.

```xaml
<toolkit:Popup>
         <Grid>
               <Grid.Resources>
                     <!-- your styles here -->
               </Grid.Resources>
         </Grid>
</toolkit:Popup>
```

**Comment by CliffAgius** [comment]
Closed as answered
